http/go/nethttp/pipeline.go

```diff
@@ -0,0 +1,54 @@
+package nethttplibrary
```

Comment:
Could we make a pass on this file for adding documentation comments for this file?

Reply:
added doc comments here and on the 2 handlers.